app/services/SubmissionService.scala

```diff
@@ -91,4 +104,62 @@ class SubmissionService @Inject()(
 
+  def getCachedData(cacheMap: CacheMap)(implicit request: JourneyRequest[_]) = {
+
+    val userInput = Map(
+      "AdditionalDeclarationType" ->
+        Json.toJson(cacheMap.getEntry[AdditionalDeclarationType]("AdditionalDeclarationType")),
+      DeclarantDetails.id -> Json.toJson(cacheMap.getEntry[DeclarantDetails](DeclarantDetails.id)),
+      ExporterDetails.id -> Json.toJson(cacheMap.getEntry[ExporterDetails](ExporterDetails.id)),
+      RepresentativeDetails.formId -> Json.toJson(
+        cacheMap.getEntry[RepresentativeDetails](RepresentativeDetails.formId)
+      ),
+      TransactionType.formId -> Json.toJson(cacheMap.getEntry[TransactionType](TransactionType.formId)),
+      CarrierDetails.id -> Json.toJson(cacheMap.getEntry[CarrierDetails](CarrierDetails.id)),
+      ConsigneeDetails.id -> Json.toJson(cacheMap.getEntry[ConsigneeDetails](ConsigneeDetails.id)),
+      GoodsLocation.formId -> Json.toJson(cacheMap.getEntry[GoodsLocation](GoodsLocation.formId)),
+      ConsignmentReferences.id -> Json.toJson(cacheMap.getEntry[ConsignmentReferences](ConsignmentReferences.id)),
+      BorderTransport.formId -> Json.toJson(cacheMap.getEntry[BorderTransport](BorderTransport.formId)),
+      TransportDetails.formId -> Json.toJson(cacheMap.getEntry[TransportDetails](TransportDetails.formId)),
+      DestinationCountries.formId -> getDestinationCountries(cacheMap),
+      DispatchLocation.formId -> Json.toJson(cacheMap.getEntry[DispatchLocation](DispatchLocation.formId)),
+      OfficeOfExit.formId -> getOfficeOfExit(cacheMap),
+      DeclarationAdditionalActorsData.formId -> Json.toJson(
+        cacheMap
+          .getEntry[DeclarationAdditionalActorsData](DeclarationAdditionalActorsData.formId)
+          .map(_.actors) getOrElse (Seq.empty)
+      ),
+      DeclarationHoldersData.formId -> Json.toJson(
+        cacheMap.getEntry[DeclarationHoldersData](DeclarationHoldersData.formId).map(_.holders) getOrElse (Seq.empty)
+      ),
+      TransportInformationContainerData.id -> Json.toJson(
+        cacheMap
+          .getEntry[TransportInformationContainerData](TransportInformationContainerData.id)
+          .map(_.containers) getOrElse (Seq.empty)
+      ),
+      TransportInformationContainerData.id -> Json.toJson(
+        cacheMap.getEntry[Seq[InternalAgencyGoodsItem]](ExportsItemsCacheIds.itemsId).getOrElse(Seq.empty)
+      ),
+      Seal.formId -> Json.toJson(cacheMap.getEntry[Seq[Seal]](Seal.formId)),
+      Document.formId -> Json.toJson(
+        cacheMap.getEntry[PreviousDocumentsData](Document.formId).map(_.documents).getOrElse(Seq.empty)
+      ),
+      TotalNumberOfItems.formId -> Json.toJson(cacheMap.getEntry[TotalNumberOfItems](TotalNumberOfItems.formId)),
+      WarehouseIdentification.formId -> Json.toJson(
+        cacheMap.getEntry[WarehouseIdentification](WarehouseIdentification.formId)
+      )
+    )
+    Json.toJson(userInput).as[JsObject]
```

Comment:
Instead of doing this (I'm not sure about this JsObject). What do you think about 
`Json.obj(Map(...))`
instead
`val userInput = Map(...)` and `Json.toJson(userInput).as[JsObject]`

Reply:
I will try

Reply:
Json.obj expecting obj(fields: (String, JsValueWrapper)*)

it would be changing from Map to tuple and the method does the something similar as below = JsObject(fields.map(f => (f._1, f._2.asInstanceOf[JsValueWrapperImpl].field)))